src/mattack_actors.cpp

```diff
@@ -799,6 +821,7 @@ bool gun_actor::call( monster &z ) const
 bool gun_actor::try_target( monster &z, Creature &target ) const
 {
     if( require_sunlight && !g->is_in_sunlight( z.pos() ) ) {
+        add_msg_debug( debugmode::DF_MATTACK, "Requires sunlight, attack failed" );
```

Comment:
I'd add a monster's name here.

Reply:
The name is printed from monmove when attempting the attack, so it should be clear in sequence (I should remove attack failed, though, that gets printed from there as well).